app/controllers/application_controller.rb

```diff
@@ -4,6 +4,7 @@ class ApplicationController < ActionController::Base
   default_form_builder GOVUKDesignSystemFormBuilder::FormBuilder
   rescue_from ActiveRecord::RecordNotFound, with: :not_found
   rescue_from Pundit::NotAuthorizedError, with: :pundit_not_found
+  rescue_from Faraday::Error, with: :postcode_api_error
```

Comment:
I think catching the error here for this particular use case is not appropriate. We need a localised error handling in the form. I'd say the `PostcodeValidator` should catch it.

Reply:
I'm not sure that is best, if postcodes.io is down, or changes urls, we can't progress the form, as we can't validate the existence of  postcodes, or do any search filtering ? If we catch the error and display "There is a problem, come back later..." we would have to rely on our exception alarm system to notify us of an issue?

Reply:
The problem is that redirecting to another page loses everything in that form and there's no way back half way through the shortlisting journey. I think it's awkward. Also this will presumably be in multiple places in the app?

Reply:
Yes I would imagine wherever we enter a postcode. If the service we use is down, we will have to plan how we handle that, if you want catch it in the form and display a "please try again in 5 mins" we can go with that for now , no worries.

Reply:
That's probably a Dimitri question 🤔